TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/NettyScanningHttpProxy.java

```diff
@@ -58,4 +58,8 @@ public void stop() throws InterruptedException {
             bossGroup.shutdownGracefully();
         }
     }
+
+    public void waitForClose() throws InterruptedException {
```

Comment:
This looks cleaner, and seems like it should work but makes me a little skeptical

Reply:
worst case scenario is that it doesn't and the process is killed.  That's pretty much where we were before too w/ the inescapable loop.